src/main/java/org/opentripplanner/street/model/vertex/StreetVertex.java

```diff
@@ -72,4 +75,29 @@ public boolean isConnectedToDriveableEdge() {
   public boolean isEligibleForCarPickupDropoff() {
     return isConnectedToDriveableEdge() && isConnectedToWalkingEdge();
   }
+
+  /**
+   * Returns the list of area stops that this vertex is inside of.
+   */
+  @Nonnull
+  public Set<AreaStop> areaStops() {
+    return Objects.requireNonNullElseGet(areaStops, Set::of);
+  }
+
+  /**
+   * Add a collection of area stops to this vertex.
+   */
+  public void addAreaStops(@Nonnull Collection<AreaStop> toBeAdded) {
+    Objects.requireNonNull(toBeAdded);
+
+    synchronized (this) {
+      if (areaStops == null) {
+        areaStops = Set.copyOf(toBeAdded);
+      } else {
+        var set = new HashSet<>(areaStops);
```

Comment:
Is there a reason for modifying `areaStops` this way? (This is creating two sets.)

Reply:
The reason is that `Set.copyOf` is immutable and therefore we need a temporary collection to add everything up and then make it immutable again at the end. (Immutable collections save memory, and for large flex zones this adds up.)

Reply:
Possibly using stream concatenation would make the intent (merging 2 sets into a new unmodifiable set) more visible :
```
  public void addAreaStops(@Nonnull Collection<AreaStop> toBeAdded) {
    Objects.requireNonNull(toBeAdded);
    synchronized (this) {
      if (areaStops == null) {
        areaStops = Set.copyOf(toBeAdded);
      } else {
        areaStops =
          Stream
            .concat(areaStops.stream(), toBeAdded.stream())
            .collect(Collectors.toUnmodifiableSet());
      }
    }
  }
```

Reply:
Done.